main.go

```diff
@@ -58,13 +70,22 @@ func portalClient() *http.Client {
 	}
 }
 
-func resetPasswords(f *excelize.File, input *Input, portal *Portal, s3Client S3ClientAPI) (err error) {
-	automatedSheet := input.AutomatedSheetName
+func resetPasswords(f *excelize.File, input *Input, portal *Portal, s3Client S3ClientAPI, env Environment) (err error) {
+	automatedSheet := input.SheetGroups[env].AutomatedSheetName
 	rows, err := f.GetRows(automatedSheet)
 	if err != nil {
 		return err
 	}
 
+	sheetName := input.SheetGroups[env].SheetName
+	mcFinUsersToPasswordRow, err := getMACFinUsers(f, input, env)
+	mfRows, err := f.GetRows(sheetName)
+	if err != nil {
+		return err
+	}
+	headerNameToXCoord := getHeaderToXCoord(mfRows[0])
```

Comment:
what does this code (and this program generally) do if one of the headers given is missing in one of the input sheets? does it fail with an error? or does it overwrite stuff in an arbitrary column? maybe we should have a test case to verify. I think the ideal behavior is that it does a check at the beginning of the program.

Reply:
I added a validateMACFinSheets() function immediately after the file is downloaded. This checks that the MACFin sheets exist and that each MACFin sheet has the expected heading in row 0. I am not checking the PasswordManager sheets since those sheets are protected and cannot be tampered with.

Reply:
I think we should check both. e.g. what if the automated spreadsheet format changes and you try to run the program with the old parameters?

Reply:
I added a function that validates the automated and MACFin sheets for each portal. I added a test case to verify each check: empty sheet, wrong number of cols, missing col.